### DISCUSSION TITLE
Not able to save the annotated data of table in gt.txt using PPOCRLabel

### DISCUSSION THREAD

**Discussion by Bishow-99** [PaddlePaddle/PaddleOCR#13509] [Q&A]
#### 运行环境 / Runtime Environment
- OS: Ubuntu 20.04
- Paddle: 2.6.0
- PaddleOCR: 2.8

Please help me to understand, how I can save the table annotated data to a gt.txt file.

**Comment by SWHL** [comment]
The PPOCRLabel project has been migrated to new repositories → [link](https://github.com/PFCCLab/PPOCRLabel)

Please move to the new PPOCR Label repo and ask questions. Thank you.

**Comment by Bishow-99** [comment]
Thank you so much @SWHL  for your immediate response,  and now  PPOCRLabel works nicely, @SWHL  could you please tell me does PaddleOCR works for an image like below
![Screenshot from 2024-07-25 23-18-05](https://github.com/user-attachments/assets/e631ebe3-6eb1-40e4-8159-cc1a3dff573e)
 We can see above it has three borderless columns and multiple rowspan. If we work then what could be our annotation and training strategy?

**Reply by SWHL** [reply]
You can try your image by this [online demo](https://aistudio.baidu.com/community/app/91661/webUI). 
If it can satisfy your command, I think you could try to train your own model by refering this [docs](https://paddlepaddle.github.io/PaddleOCR/en/ppstructure/model_train/train_table.html#43-calculate-teds)

**Comment by Bishow-99** [comment]
Hello @SWHL I have given the below image to [online demo](https://aistudio.baidu.com/community/app/91661/webUI).
![Screenshot from 2024-07-26 15-02-24](https://github.com/user-attachments/assets/342e2f47-c74d-42dd-8fb3-c9aa3c132bea)

I got the below one
![Screenshot from 2024-07-26 15-14-39](https://github.com/user-attachments/assets/7a29a023-2661-424b-8347-ad98f2ffc839)

I understand that we need to train our own model for accurate results. But even after following this [docs](https://paddlepaddle.github.io/PaddleOCR/en/ppstructure/model_train/train_table.html#43-calculate-teds)  and training the own model I am getting still inaccurate results. Please click this  #13499 where I have described all the processes that I have done to train the table structure recognition model. Although, It correctly identifies HTML structures, but cell coordinates are in the wrong position. So, could you please give me  how can we handle these things.

**Reply by SWHL** [reply]
This is usually caused by too little training data. How much data did you use to train this model?

**Reply by Bishow-99** [reply]
Hello @SWHL , I was eagerly waiting for you rresponse,  well to evaluate the performance of PaddleOCR, I used a single image and annotated it in ten different ways by varying the bounding box localization. I then replicated these ten annotations to create a dataset of 500 images to 4000 images . This dataset was used for training, while the same image with different localizations was used for validation.

My goal was to determine if PaddleOCR could work with this data without extensive initial annotation. However, the accuracy remains unsatisfactory. Although PaddleOCR successfully identifies the HTML structure's shape, the cell locations are mismatched.

I am following annotation strategies of table stucture recoginiton from PPOCRLabel but I need guidance on how to properly annotate multi-line row spans to improve the accuracy. Here is the current output, where the HTML structure is identified correctly, but the cells are positioned incorrectly:


![image](https://github.com/user-attachments/assets/2bc99ceb-bcee-4e5f-8487-9a33299a5641)

Could you please provide guidance on how to annotate and train the table recognition model in above images?

**Reply by SWHL** [reply]
I think you use too little training data. 
I suggest you pre-train with some public table recognition datasets, such as [PubTabNet](https://developer.ibm.com/exchanges/data/all/pubtabnet/), and then fine-tune with your own labeled data. This may give better results.

**Reply by Bishow-99** [reply]
@SWHL, I've been fine-tuning the PaddleOCR table structure recognition model, which is already pre-trained on datasets like PubTabNet. Given this, do you think it's sufficient, or should I consider training the model from scratch and then fine-tuning it with my own labeled data? I am still confused.

**Reply by SWHL** [reply]
I think the training data is not enough, you need to get more data for training the pre-trained model.

**Reply by Bishow-99** [reply]
Okay @SWHL in conclusion, you're suggesting that I should increase the size of my dataset and use the PaddleOCR pre-trained table structure model to fine-tune with my own data for better performance. Is that correct?

At the same time could you please check my annotation for the particular image  #13499

**Reply by SWHL** [reply]
> Okay @SWHL in conclusion, you're suggesting that I should increase the size of my dataset and use the PaddleOCR pre-trained table structure model to fine-tune with my own data for better performance. Is that correct?

Yes

> At the same time could you please check my annotation for the particular image https://github.com/PaddlePaddle/PaddleOCR/discussions/13499

I think it's no problem.

**Comment by Bishow-99** [comment]
@SWHL Thank you so much for these information

**Comment by 0225pang** [comment]
> #### 问题描述 / 我正在使用 PPOCRLabel 进行表格注释，我在 PPOCRLabel 工具中成功注释了表格数据，但很难将表格的注释文件保存到gt.txt。虽然它在我的系统上创建了gt.txt，但即使我在 PPOCRLabel 工具上成功保存了带注释的数据，它在内部仍然为空。
> #### 运行环境 / Runtime Environment
> * 操作系统： Ubuntu 20.04
> * 拨片：2.6.0
> * PaddleOCR：2.8
> 
> 请帮助我了解如何将表注释数据保存到 gt.txt 文件中。

请问您是怎么解决的？
